eth/beacon/types/blocks.py

```diff
@@ -111,3 +113,17 @@ def root(self) -> Hash32:
     @property
     def num_attestations(self) -> int:
         return len(self.body.attestations)
+
+    @classmethod
+    def get_block_without_signature_root(
+            cls,
+            block: 'BaseBeaconBlock') -> Hash32:
+        return cls(
```

Comment:
Ah right, using `copy` looks more concise. 👍 
If we use `copy`, it could be a `property`:
```python
    @property
    def block_without_signature_root(self) -> Hash32:
        return self.copy(
            signature=EMPTY_SIGNATURE,
        ).root
```

it assumes the type of "block_without_signature" is as same as the given `block`